docs/section4/4-3.md

```diff
@@ -0,0 +1,104 @@
+
+    - Get the analogous Zone ID in the P_RMR: `zone_p = match_data_element(P_RMR,Zones,zone_u.id)`
+        - Check if there are terminal units associated with the zone, if not skip to if statements based on if there are heating and cooling systems: `if len(zone_p.terminals) != 0:`
+            - For each terminal unit associated with each zone: `for terminal_p in zone_p:`
+                - if reheat_source is not null then set heating exists boolean variable equal to true: `if terminal_p.reheat_source != "None":`
+                    - Set heating system exists boolean variable equal to true: `heating_system_exists_p = TRUE`
+                - if heat_capacity is not zero then set heating exists boolean variable equal to true: `if terminal_p.heat_capacity != 0:`
+                    - Set heating system exists boolean variable equal to true: `heating_system_exists_p = TRUE`    
+                - Get the served_by_heating_ventilation_air_conditioning_systems for each terminal: `heating_ventilation_air_conditioning_systems_p = terminal_p.served_by_heating_ventilation_air_conditioning_systems`
+                - Add to the list of heating_ventilation_air_conditioning_systems_p as the code loops through the terminal units: `heating_ventilation_air_conditioning_systems_list_p = heating_ventilation_air_conditioning_systems_list_p.append(heating_ventilation_air_conditioning_systems_p)`                    
+            - Convert the list of heating_ventilation_air_conditioning_systems_list_p associated with the zone to a set and the back to a list to eliminate duplicates: `heating_ventilation_air_conditioning_systems_list_p = list(set(heating_ventilation_air_conditioning_systems_list_p))`
+                - For each hvac_p in heating_ventilation_air_conditioning_systems_list_p: `for hvac_p in heating_ventilation_air_conditioning_systems_list_p:`
+                    - For each heating system in hvac_p: `for heating_system_p in hvac_p.heating_system:`
+                        - if heating_system_type is not "None": `if heating_system_p.heating_system_type != "None":`
+                            - Set heating system exists boolean variable equal to true: `heating_system_exists_p = TRUE`
+                    - For each cooling system in hvac_p: `for cooling_system_p in hvac_p.cooling_system:`
+                        - if cooling_system_type is not "None": `if cooling_system_p.cooling_system_type != "None":`
+                            - Set cooling system exists boolean variable equal to true: `cooling_system_exists_p = TRUE`
+                    - For each preheat system in hvac_p: `for preheat_system_p in hvac_p.preheat_system:`
+                        - if preheat_system_type is not "None": `if preheat_system_p.heating_system_type != "None":`
+                            - Set heating system exists boolean variable equal to true: `heating_system_exists_p = TRUE`  
+        
```

Comment:
@claperle and @mkarpman i now understand why the applicability checks for 10-10 and 10-11 have been added. this is quite detailed and very well written- thank you!

Does it make sense to convert the check for if system is existing/has been designed to a function? There might be more rules that need this check.

Reply:
Yup, done!